### PR TITLE
[FEATURE] Implement "is_close()" for making equality comparisons "reasonably close" for each ExecutionEngine subclass

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Implementation of `is_close()` for all `ExecutionEngine` types.
* Existing tests must pass.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1106
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py [hunk 1]
```diff
@@ -722,7 +722,7 @@ def _get_bootstrap_estimate(
             float
         ] = get_parameter_value_and_validate_return_type(
             domain=domain,
-            parameter_reference=kwargs.get("float"),
+            parameter_reference=kwargs.get("quantile_bias_std_error_ratio_threshold"),
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/parameter_builder/numeric_metric_range_multi_batch_parameter_builder.py:725]: From our discussion : this was a typo, which our test caught :)

**Review by Shinnnyshinshin — APPROVED** [review]
really appreciate you digging into this. Thank you. LGTM!

File: great_expectations/util.py [hunk 2]
```diff
@@ -1342,18 +1342,15 @@ def isclose(
...
+        isinstance(operand_a, datetime.datetime)
+        and isinstance(operand_b, datetime.datetime)
+    ):
+        return operand_a == operand_b
 
     return np.isclose(
-        a=operand_a_as_number,
-        b=operand_b_as_number,
+        a=np.float64(operand_a),
+        b=np.float64(operand_b),
```
Comment (NathanFarmer) [inline great_expectations/util.py:1353]: I think `np.float64` has been deprecated in favor of `float`.

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Comment by alexsherstinsky** [comment]
Known unrelated tests failures:
```
FAILED tests/data_context/test_data_context_variables.py::test_cloud_enabled_data_context_variables_e2e
ERROR tests/data_context/test_data_context_variables.py::test_cloud_data_context_variables_successfully_hits_cloud_endpoint
```

**Pull request merged**